src/core/electrostatics_magnetostatics/p3m-dipolar.hpp

```diff
@@ -221,11 +221,12 @@ void dp3m_shrink_wrap_dipole_grid(int n_dipoles);
 /** Calculate real space contribution of p3m dipolar pair forces and torques.
  *  If NPT is compiled in, it returns the energy, which is needed for NPT.
  */
-inline double dp3m_add_pair_force(Particle *p1, Particle *p2,
-                                  Utils::Vector3d const &d, double dist2,
-                                  double dist, Utils::Vector3d &force) {
+inline std::tuple<double, Utils::Vector3d, Utils::Vector3d, Utils::Vector3d>
+dp3m_add_pair_force(Particle const *const p1, Particle const *const p2,
+                    Utils::Vector3d const &d, double dist2, double dist) {
```

Comment:
Function needs to be renamed from `add` to `calc`.

Reply:
i'd suggest to drop both: ```dp3m_pair_force```

Reply:
Thanks! I've now removed the `add` part.